### PR TITLE
Add Model Generation Script

### PR DESCRIPTION
**Pull request by kcotugno**
This PR adds support for generating the gem's Models using the Intacct's [published schema](https://raw.githubusercontent.com/Intacct/intacct_dtd/master/v3.0/api.xsd). It supports:
* Custom fields
* Dates
* Required/optional attributes
* Complex attributes assuming the sub-objects responds to methods with the names of the sub-attributes
* Retains the distinctions between create and update fields

### CODE REVIEW

File: bin/sync_schema [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+  end
+
+  def to_ruby(action = "create")
+    case data_type
+    when :custom_fields
+      custom_field_ruby(action)
+    when :date
+      append_presence_check("xml.#{xml_name} attributes.#{name}.strftime(\"%m/%d/%Y\")")
+    # when :struct
+      # binding.pry
```
Comment (b6b) [inline bin/sync_schema]: 🤔

File: bin/sync_schema [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+  @update_names = @update_attributes.map { |attr| attr.name.to_sym }
+
+  if @identical = @create_names == @update_names
+    @update_names = "CREATE_KEYS"
+  end
+
+  File.write(File.expand_path("../lib/intacct/models/#{object}.rb", __dir__), ERB.new(@template, trim_mode: "-").result)
+end
+
+%w[project task].each { |object| process_template(object) }
```
Comment (b6b) [inline bin/sync_schema]: Would it make sense to break these out into a constant or somewhere more visible?

**Review by b6b — COMMENTED** [review]
This is sweet. I think I get the gist of it but can you add some details to the description?

File: bin/sync_schema (hunk 1 above)
Comment (kcotugno) [reply bin/sync_schema]: 😬

File: lib/intacct/models/project.rb [hunk 3]
```diff
@@ -1,79 +1,102 @@
 module Intacct
   module Models
     class Project < Intacct::Base
+      CREATE_KEYS = [:projectid, :name, :description, :parentid, :invoicewithparent, :projectcategory, :projecttype, :projectstatus, :customerid, :managerid, :custuserid, :salescontactid, :begindate, :enddate, :departmentid, :locationid, :classid, :currency, :billingtype, :termname, :docnumber, :billto, :shipto, :contactinfo, :sonumber, :ponumber, :poamount, :pqnumber, :budgetamount, :budgetedcost, :budgetqty, :userrestrictions, :obspercentcomplete, :budgetid, :billingrate, :billingpricing, :expenserate, :expensepricing, :poaprate, :poappricing, :status, :supdocid, :invoicemessage, :invoicecurrency, :projectresources, :billingovermax, :excludeexpenses, :customfields].freeze
```
Comment (b6b) [inline lib/intacct/models/project.rb]: Super tiny nitpick but can this be multi-lined to be a bit more readable?

File: lib/intacct/models/task.rb [hunk 4]
```diff
@@ -1,56 +1,116 @@
 module Intacct
   module Models
     class Task < Intacct::Base
+      CREATE_KEYS = [:taskname, :projectid, :taskid, :pbegindate, :penddate, :itemid, :billable, :taskdescription, :ismilestone, :utilized, :priority, :taskno, :taskstatus, :parentkey, :parenttaskname, :budgetqty, :estqty, :timetype, :obspercentcomplete, :taskresources, :customfields, :classid, :supdocid, :dependentonkey, :dependentonname].freeze
```
Comment (b6b) [inline lib/intacct/models/task.rb]: Ditto on multi-lining.

**Review by b6b — APPROVED** [review]
LGTM!

**Review by b6b — APPROVED** [review]
LGTM!

**Pull request merged**